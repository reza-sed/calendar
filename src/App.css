* {
  box-sizing: border-box;
}

body {
  font-family: Roboto, -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto,
    Oxygen, Ubuntu, Cantarell, "Open Sans", "Helvetica Neue", sans-serif;
  background: var(--color-secondary);
}

/* App global color pallete */
:root {
  --color-primary: #aeaeae;
  --color-secondary: #e4e4e4;
  --color-accent: #d4665b;
  /* events global color set */
  --color-event-success: #5cb058;
  --color-event-danger: #b02942;
  --color-event-warn: #d35c27;
}

/* define varous events related type colors */
.event--success {
  fill: var(--color-event-success);
}

.event--danger {
  fill: var(--color-event-danger);
}

.event--warn {
  fill: var(--color-event-warn);
}
