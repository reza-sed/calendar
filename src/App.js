import "./App.css";
import Calendar from "./component/calendar";

// events status
export const eventStatus = {
  success: "event--success",
  danger: "event--danger",
  warn: "event--warn",
};
// events structure
const events = [
  { id: 1, desc: "event 1", status: eventStatus.success, date: "2021-07-01" },
  { id: 2, desc: "event 2", status: eventStatus.danger, date: "2021-07-01" },
  { id: 3, desc: "event 3", status: eventStatus.success, date: "2021-07-01" },
  { id: 4, desc: "event 4", status: eventStatus.warn, date: "2021-07-02" },
];

function App() {
  return (
    <div className="App">
      <Calendar month={6} year={2021} events={events} />
    </div>
  );
}

export default App;
