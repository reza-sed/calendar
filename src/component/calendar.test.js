import React from "react";
import { cleanup, fireEvent, render } from "@testing-library/react";
import renderer from "react-test-renderer";
import Calendar from "./Calendar";
import { eventStatus } from "../App";

afterEach(cleanup);

const calendarElement = (
  <Calendar
    year={2021}
    month={0}
    events={[{ date: "2021-01-01", status: eventStatus.success }]}
  />
);

describe("Calendar", () => {
  it("change current month after click next and prev", () => {
    const { container } = render(calendarElement);
    // check prev call
    const prevArrow = container.querySelector("#prev");
    fireEvent.click(prevArrow);
    let calendarTitleElement = container.querySelector("#monthTitle");
    expect(calendarTitleElement.textContent).toBe("December 2020");
    // check next call
    const nextArrow = container.querySelector("#next");
    fireEvent.click(nextArrow);
    fireEvent.click(nextArrow);
    calendarTitleElement = container.querySelector("#monthTitle");
    expect(calendarTitleElement.textContent).toBe("February 2021");
  });

  it("check if calendar is rendered", () => {
    const { container } = render(calendarElement);
    const header = container.querySelector("h3");
    expect(header.textContent).toBe("Calandar");
  });

  it("snapshot", () => {
    const component = renderer.create(calendarElement);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
