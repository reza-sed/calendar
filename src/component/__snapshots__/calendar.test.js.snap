// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calendar snapshot 1`] = `
<section
  className="calendar"
>
  <h3>
    Calandar
  </h3>
  <div
    className="panel"
  >
    <span
      className="panel__title"
      id="monthTitle"
    >
      January 2021
    </span>
    <span
      className="panel__navigation"
    >
      <span
        className="icon"
        id="prev"
        onClick={[Function]}
      >
        <svg
          className="icon__element"
        >
          <use
            xlinkHref="/images/icons.svg#prev"
          />
        </svg>
      </span>
      <span
        className="icon"
        id="next"
        onClick={[Function]}
      >
        <svg
          className="icon__element"
        >
          <use
            xlinkHref="/images/icons.svg#next"
          />
        </svg>
      </span>
    </span>
  </div>
  <section
    className="grid"
  >
    <div
      className="grid__header"
    >
      Mon
    </div>
    <div
      className="grid__header"
    >
      Tue
    </div>
    <div
      className="grid__header"
    >
      Wed
    </div>
    <div
      className="grid__header"
    >
      Thu
    </div>
    <div
      className="grid__header"
    >
      Fri
    </div>
    <div
      className="grid__header"
    >
      Sat
    </div>
    <div
      className="grid__header"
    >
      Sun
    </div>
    <article
      className="day outmonth"
    >
      <h4>
        28
      </h4>
      <p />
    </article>
    <article
      className="day outmonth"
    >
      <h4>
        29
      </h4>
      <p />
    </article>
    <article
      className="day outmonth"
    >
      <h4>
        30
      </h4>
      <p />
    </article>
    <article
      className="day outmonth"
    >
      <h4>
        31
      </h4>
      <p />
    </article>
    <article
      className="day inmonth"
    >
      <h4>
        1
      </h4>
      <p>
        <span
          className="icon"
          id="circle"
        >
          <svg
            className="icon__element event--success"
          >
            <use
              xlinkHref="/images/icons.svg#circle"
            />
          </svg>
        </span>
      </p>
    </article>
    <article
      className="day inmonth"
    >
      <h4>
        2
      </h4>
      <p />
    </article>
    <article
      className="day inmonth"
    >
      <h4>
        3
      </h4>
      <p />
    </article>
    <article
      className="day inmonth"
    >
      <h4>
        4
      </h4>
      <p />
    </article>
    <article
      className="day inmonth"
    >
      <h4>
        5
      </h4>
      <p />
    </article>
    <article
      className="day inmonth"
    >
      <h4>
        6
      </h4>
      <p />
    </article>
    <article
      className="day inmonth"
    >
      <h4>
        7
      </h4>
      <p />
    </article>
    <article
      className="day inmonth"
    >
      <h4>
        8
      </h4>
      <p />
    </article>
    <article
      className="day inmonth"
    >
      <h4>
        9
      </h4>
      <p />
    </article>
    <article
      className="day inmonth"
    >
      <h4>
        10
      </h4>
      <p />
    </article>
    <article
      className="day inmonth"
    >
      <h4>
        11
      </h4>
      <p />
    </article>
    <article
      className="day inmonth"
    >
      <h4>
        12
      </h4>
      <p />
    </article>
    <article
      className="day inmonth"
    >
      <h4>
        13
      </h4>
      <p />
    </article>
    <article
      className="day inmonth"
    >
      <h4>
        14
      </h4>
      <p />
    </article>
    <article
      className="day inmonth"
    >
      <h4>
        15
      </h4>
      <p />
    </article>
    <article
      className="day inmonth"
    >
      <h4>
        16
      </h4>
      <p />
    </article>
    <article
      className="day inmonth"
    >
      <h4>
        17
      </h4>
      <p />
    </article>
    <article
      className="day inmonth"
    >
      <h4>
        18
      </h4>
      <p />
    </article>
    <article
      className="day inmonth"
    >
      <h4>
        19
      </h4>
      <p />
    </article>
    <article
      className="day inmonth"
    >
      <h4>
        20
      </h4>
      <p />
    </article>
    <article
      className="day inmonth"
    >
      <h4>
        21
      </h4>
      <p />
    </article>
    <article
      className="day inmonth"
    >
      <h4>
        22
      </h4>
      <p />
    </article>
    <article
      className="day inmonth"
    >
      <h4>
        23
      </h4>
      <p />
    </article>
    <article
      className="day inmonth"
    >
      <h4>
        24
      </h4>
      <p />
    </article>
    <article
      className="day inmonth"
    >
      <h4>
        25
      </h4>
      <p />
    </article>
    <article
      className="day inmonth"
    >
      <h4>
        26
      </h4>
      <p />
    </article>
    <article
      className="day inmonth"
    >
      <h4>
        27
      </h4>
      <p />
    </article>
    <article
      className="day inmonth"
    >
      <h4>
        28
      </h4>
      <p />
    </article>
    <article
      className="day inmonth"
    >
      <h4>
        29
      </h4>
      <p />
    </article>
    <article
      className="day inmonth"
    >
      <h4>
        30
      </h4>
      <p />
    </article>
    <article
      className="day inmonth"
    >
      <h4>
        31
      </h4>
      <p />
    </article>
    <article
      className="day outmonth"
    >
      <h4>
        1
      </h4>
      <p />
    </article>
    <article
      className="day outmonth"
    >
      <h4>
        2
      </h4>
      <p />
    </article>
    <article
      className="day outmonth"
    >
      <h4>
        3
      </h4>
      <p />
    </article>
    <article
      className="day outmonth"
    >
      <h4>
        4
      </h4>
      <p />
    </article>
    <article
      className="day outmonth"
    >
      <h4>
        5
      </h4>
      <p />
    </article>
    <article
      className="day outmonth"
    >
      <h4>
        6
      </h4>
      <p />
    </article>
    <article
      className="day outmonth"
    >
      <h4>
        7
      </h4>
      <p />
    </article>
  </section>
</section>
`;
